### DISCUSSION TITLE
How can I accept multiple UUID url params?

### DISCUSSION THREAD

**Discussion by ElginCahangirov** [carltongibson/django-filter#1501] [Q&A]
I defined my field as:

`region = UUIDFilter(method="filter_by_region")`

When client send urls params like `?region=f157cb5a-7cb0-4672-82b3-3d5e381b6c60&region=f157cb5a-7cb0-4672-82b3-3d5e381b6c61`, I got only the last value in `value` parameter of `filter_by_region` method, but I need all of them as a list. What is the best way of handling this use case? I would expect parameter like `many=True` for filter object or some kind of wrapper that accepts filter object for representing array of objects (e.g. `ArrayFilter(UUIDFilter(...))`). But I couldn't find any of these in the documentation and also any other possible solution.

How would you implement this?

**Comment by carltongibson** [comment]
Take a look at `MultipleChoiceFilter` and `ModelMultipleChoiceFilter` https://django-filter.readthedocs.io/en/stable/ref/filters.html#modelmultiplechoicefilter

At the bottom level they both use a widget that calls [QueryDict.getlist()](https://docs.djangoproject.com/en/4.0/ref/request-response/#django.http.QueryDict.getlist) to fetch a list of values, rather than just a single value, from the query string.

**Reply by ElginCahangirov** [reply]
@carltongibson but I don't intend to introduce any choice. I mean client should be able to send UUID values as many as it wishes, which are not related to any database model or list of choices.

**Reply by carltongibson** [reply]
Which is why I pointed you to the QueryDict docs. 

You'll need a widget that uses `getlist`, a Form Field that validates and maps strings to UUID instances, and then a Filter that uses that Field. (It sounds a lot but each class is quite limited in scope. You should be able to follow the existing examples in package.)

**Reply by ElginCahangirov** [reply]
@carltongibson should I do the same for `CharFilter` too, if want to accept list of strings? I think this is frequently used pattern in REST APIs, so it is weird why there's no builtin support.
